include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -31,6 +31,7 @@
 #include "parallel_backend.h"
 #include "parallel_impl.h"
 #include "iterator_impl.h"
+#include "../functional"
```

Comment:
Should we still have `#include <functional>` above?

Reply:
ok

Reply:
Actually, 
I pretty sure that something from `https://en.cppreference.com/w/cpp/header/functional` is used within the next set of header files:
```
#include "algorithm_fwd.h"

#include "execution_impl.h"
#include "memory_impl.h"
#include "parallel_backend_utils.h"
#include "unseq_backend_simd.h"

#include "parallel_backend.h"
#include "parallel_impl.h"
#include "iterator_impl.h"
```
And I am not sure that we move above `#include "../functional"` because there might be the "cross dependencies".
I believe there is a reason why certain including order is.

Reply:
At least you may check what happens if you simple remove `#include <functional>` in this file.

Reply:
Thank you for check, @MikeDvorskiy.
Let's resolve this conversation.